airgen/src/lib.rs

```diff
@@ -178,14 +174,25 @@ fn resolve_submachine_arg(
     }
 }
 
-fn utility_functions(asm_file: AnalysisASMFile) -> BTreeMap<AbsoluteSymbolPath, TypeOrExpression> {
+fn utility_functions(asm_file: AnalysisASMFile) -> BTreeMap<AbsoluteSymbolPath, Vec<PilStatement>> {
     asm_file
-        .items
+        .modules
         .into_iter()
-        .filter_map(|(n, v)| match v {
-            Item::Expression(e) => Some((n, TypeOrExpression::Expression(e))),
-            Item::TypeDeclaration(type_decl) => Some((n, TypeOrExpression::Type(type_decl))),
-            _ => None,
+        .map(|(module_path, module)| {
+            (
+                module_path,
+                module
+                    .into_inner()
+                    .1
+                    .into_iter()
+                    .filter(|s| {
+                        matches!(
+                            s,
+                            PilStatement::EnumDeclaration(..) | PilStatement::LetStatement(..)
```

Comment:
I already envision spending quite some time trying to figure out why something is missing from the output. Is there a safer way to do this check so that we add variants here if the code is extended?

Reply:
using an exhaustive match instead